Sources/AWSLambdaRuntimeCore/LambdaContext.swift

```diff
@@ -93,20 +172,20 @@ extension Lambda {
                       logger: Logger,
                       eventLoop: EventLoop,
                       allocator: ByteBufferAllocator) {
-            self.requestID = requestID
-            self.traceID = traceID
-            self.invokedFunctionARN = invokedFunctionARN
-            self.cognitoIdentity = cognitoIdentity
-            self.clientContext = clientContext
-            self.deadline = deadline
-            // utility
-            self.eventLoop = eventLoop
-            self.allocator = allocator
-            // mutate logger with context
-            var logger = logger
-            logger[metadataKey: "awsRequestID"] = .string(requestID)
-            logger[metadataKey: "awsTraceID"] = .string(traceID)
-            self.logger = logger
+            var baggage = Baggage.background
+            baggage.lambdaRequestID = requestID
+            baggage.lambdaTraceID = traceID
+            self.storage = _Storage(
+                baggage: baggage,
+                invokedFunctionARN: invokedFunctionARN,
+                deadline: deadline,
+                cognitoIdentity: cognitoIdentity,
+                clientContext: clientContext,
+                // utility
+                _logger: logger,
+                eventLoop: eventLoop,
+                allocator: allocator
```

Comment:
iirc evenloop and allocator are immutable and dont need to be at the storage level imo

Reply:
or is this or allow them to be as part of the by-ref trick?

Reply:
Yeah this is to keep the outer Context type small -- just a single pointer, otherwise we'd grow the struct and each time it gets `let` or passed somewhere those references would be copied -- this way, everything in storage, nothing gets copied other than the single pointer just when passing the context around.

Reply:
please correct me if i am wrong, as far as I understand it:

1. everything (new storage object) is copied every time baggage is updated, which will happen every time new span is created = often

2. this does implement copy on write however changing of the storage is not thread safe which I think is not a nice API for user who would need to sync the changes himself